### PR TITLE
allow file based secrets on pods

### PR DESCRIPTION
**Pull request by fatz**
adding Secret for volumes to enable file based secrets support

### CODE REVIEW

File: volume.go [hunk 1]
```diff
@@ -38,6 +39,14 @@ func NewPodVolume(name, path string) *PodVolume {
 	}
 }
 
+// NewPodVolume creates a new PodVolume for file based secrets
+func NewPodVolumeSecret(name, secretPath string) *PodVolume {
```
Comment (timoreimann) [inline volume.go]: Would we also need/want `PodVolume.Host` in the case of defining a pod volume for file-based secrets, or is that optional?

Comment (fatz) [reply volume.go]: its not even supported. https://mesosphere.github.io/marathon/docs/secrets.html#example-configuration-of-file-based-secrets-for-pod-definitions

The flow is:
1. Define `PodVolume` to provide the secret
2. Define PodVolumeMount us use the secret and specify the container path.

Host is in this case not relevant. Furthermore AFAIK it must not be specified.

Comment (timoreimann) [reply volume.go]: Gotcha, thanks for explaining.

**Comment by timoreimann** [comment]
@fatz could you rebase / update the branch so that we can merge?

**Comment by fatz** [comment]
@timoreimann updated and tests where running through

**Pull request merged**